### PR TITLE
Azure image plugin fixes

### LINKED ISSUE #446: Image ID doesn't working properly when creating a virtual machine

### PR DESCRIPTION
**Pull request by jadsonluan**
This pull request fixes #446 

The solution was encoding/decoding the image id, this way we can extract the information we need just decoding it.

### CODE REVIEW

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java [hunk 1]
```diff
@@ -70,5 +92,76 @@ public void testBuildAzureVirtualMachineImageBySuccessfully() {
...
+        Assert.assertNotEquals(encodedId, id);
+    }
+
+    // test case: When calling encodeImageId method and a UnsupportedEncodingException
+    // occurs, it must verify if it rethrow a UnexpectedException
+    @Test
+    public void testEncodeImageIdFail() throws UnsupportedEncodingException, UnexpectedException {
+        // set up
+        PowerMockito.mockStatic(URLEncoder.class);
+        String id = "publisher@#offer@#sku";
```
Comment (fgan1) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java:115]: Use the Separator constant.

Comment (fgan1) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java:115]: Use it the same idea in all test class, so you must check other tests.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        AzureImageOperationUtil.class,
+        URLDecoder.class,
+        URLEncoder.class
+})
 public class AzureImageOperationUtilTest {
```
Comment (fgan1) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java:28]: You could add a method that tests the encode and decode method in the same test case.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/image/AzureImageOperation.java [hunk 3]
```diff
@@ -43,7 +44,7 @@ public AzureImageOperation(String region) {
                         String offerName = offer.name();
                         String skuName = sku.name();
                         ImageSummary imageSummary = AzureImageOperationUtil.buildImageSummaryBy(publisherName, offerName, skuName);
-                        String id = UUID.randomUUID().toString();
+                        String id = AzureImageOperationUtil.encodeImageId(imageSummary.getId());
```
Comment (fgan1) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/image/AzureImageOperation.java:47]: If this line throws an exception, the whole method will fail. Is it what we want?

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java (hunk 1 above)
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java]: Done.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java (hunk 2 above)
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java:28]: Done.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/image/AzureImageOperation.java (hunk 3 above)
Comment (jadsonluan) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/image/AzureImageOperation.java]: No. It's fixed now.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java (hunk 2 above)
Comment (jadsonluan) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureImageOperationUtilTest.java:28]: Done.

File: src/main/java/cloud/fogbow/ras/constants/Messages.java [hunk 4]
```diff
@@ -237,5 +237,6 @@
         public static final String ERROR_CREATE_VNET_ASYNC_BEHAVIOUR = "Error while creating virtual network asynchronously.";
         public static final String ERROR_DELETE_VNET_ASYNC_BEHAVIOUR = "Error while deleting virtual network asynchronously.";
         public static final String ERROR_DELETE_SECURITY_GROUP_ASYNC_BEHAVIOUR = "Error while deleting security group asynchronously.";
+        public static final String ERROR_WHILE_LOADING_IMAGE_S = "Error while loading the following image: %s";
```
Comment (fgan1) [inline src/main/java/cloud/fogbow/ras/constants/Messages.java:240]: Add a "." at the end of this message.

**Pull request merged**